src/Shadow.spec.ts

```diff
@@ -0,0 +1,11 @@
+import '../jest.extend';
+import { Shadow } from './Shadow';
+
+describe('Shadow', () => {
+    it('fromObject', async () => {
+        const shadow = await Shadow.fromObject({});
+        expect(shadow).toMatchObjectSnapshot();
+        expect(shadow).toMatchObjectSnapshot({ includeDefaultValues: false });
+    });
+
```

Comment:
While this test works, it won't catch a regression.
I would suggest we add a test that tests that then is defined.

```ts
it('fromObject is a promise', async () => {
        const promise = Shadow.fromObject({});
        expect(promise.then).toBeDefined(); // <--- or whatever methods exist to test the existance of the then
    });
```

Reply:
done